src/main/webapp/script.js

```diff
@@ -50,9 +50,20 @@ function createDOMListing(listing) {
     body.querySelector('.body-desc').innerText = listing.description;
     body.querySelector('#image-Url').src = listing.imageUrl;
 
-    // Fill in hidden key data
-    let hiddenKey = contactForm.querySelector('input[name="key"]');
-    hiddenKey.setAttribute('value', listing.uniqueKey);
+    // User's own post, remove contact button HTML / form
+    if (currentUser.info.userId === listing.userId) {
```

Comment:
Ideally, in addition to client check, servlet should also verify that the user deletes its own listing. Otherwise any user can delete any listing by crafting request with the right datastore key and sending it with curl.

Reply:
I did not even think about this, but I definitely agree. Just added a commit that adds in this functionality. 👍